### PR TITLE
Flake fixes for GLM module

### PR DESCRIPTION
**Pull request by rob-luke**
Following #2915 and #2916 I noticed that flake is only run on changes to code (I think?). So I manually ran a flake check locally on the GLM module and then manually fixed most errors. I didn't use an automated approach, I can if you prefer.

No hard feeling if you want to reject this PR, I didn't ask in advance, and it touches a lot of code, so is probably risky.

TODO:

- [x] Check the doc strings render correctly for the long multiline type specifications

### CODE REVIEW

File: nilearn/glm/tests/test_dmtx.py [hunk 1]
```diff
@@ -12,15 +12,11 @@
...
-    assert_equal)
+    assert_almost_equal, assert_array_equal, assert_array_almost_equal)
 
-from nilearn.glm.first_level.design_matrix import (_convolve_regressors,
-                                                   _cosine_drift,
-                                                   check_design_matrix,
-                                                   make_first_level_design_matrix,
-                                                   make_second_level_design_matrix,
-                                                   )
+from nilearn.glm.first_level.design_matrix import \
```
Comment (rob-luke) [inline nilearn/glm/tests/test_dmtx.py:17]: I changed the style here. Would you prefer one import per line?

File: nilearn/glm/first_level/design_matrix.py [hunk 2]
```diff
@@ -281,7 +281,8 @@ def make_first_level_design_matrix(
         Particular attention should be given to the 'trial_type' key
         which defines the different conditions in the experimental paradigm.
 
-    hrf_model : {'glover', 'spm', 'spm + derivative', 'spm + derivative + dispersion',
+    hrf_model : {'glover', 'spm', 'spm + derivative',
+        'spm + derivative + dispersion',
         'glover + derivative', 'glover + derivative + dispersion',
         'fir', None}, optional
```
Comment (tsalo) [inline nilearn/glm/first_level/design_matrix.py:287]: It looks like this still doesn't render correctly:

image

File: nilearn/glm/first_level/design_matrix.py [hunk 3]
```diff
@@ -301,7 +302,8 @@ def make_first_level_design_matrix(
         In case of FIR design, yields the array of delays used in the FIR
         model (in scans). Default=[0].
 
-    add_regs : array of shape(n_frames, n_add_reg) or pandas DataFrame, optional
+    add_regs : array of shape(n_frames, n_add_reg) or
+               pandas DataFrame, optional
```
Comment (tsalo) [inline nilearn/glm/first_level/design_matrix.py:306]: Splitting the data type across lines like this doesn't seem to render as expected:

image

**Comment by rob-luke** [comment]
Thanks @tsalo. I've looked up how scikit-learn does this [example](https://github.com/scikit-learn/scikit-learn/blob/d6735f4851d828984a0517de954b9b88c74919fe/sklearn/decomposition/_nmf.py#L756-L757) and will try that approach. I will try a few things and ping you when its rendering nicely.

**Comment by rob-luke** [comment]
Ok I think the doc strings are now rendering correctly. See:

* https://7867-1235740-gh.circle-artifacts.com/0/dev/modules/generated/nilearn.glm.first_level.make_first_level_design_matrix.html#nilearn.glm.first_level.make_first_level_design_matrix
* https://7867-1235740-gh.circle-artifacts.com/0/dev/modules/generated/nilearn.glm.first_level.FirstLevelModel.html#nilearn.glm.first_level.FirstLevelModel

Thanks for the review @tsalo Please let me know if you want any other changes made.

File: nilearn/glm/first_level/experimental_paradigm.py [hunk 4]
```diff
@@ -86,14 +87,13 @@ def check_events(events):
...
+                       "data will be ignored.").format(unexpected_column))
 
     # Make sure we have a numeric type for duration
     if not is_numeric_dtype(events_copy['duration']):
         try:
             events_copy = events_copy.astype({'duration': float})
-        except:
+        except ValueError:
             raise ValueError("Could not cast duration to float "
                              "in events data.")
```
Comment (tsalo) [inline nilearn/glm/first_level/experimental_paradigm.py:98]: I haven't checked this. Is the original error definitely a ValueError?

File: nilearn/glm/tests/test_dmtx.py (hunk 1 above)
Comment (tsalo) [reply nilearn/glm/tests/test_dmtx.py:17]: I don't think this is covered in the style guide, but I personally prefer `black`'s style. 

So it would be more like

```python
from X import (
    x,
    y,
    z,
)
```

**Review by tsalo — CHANGES_REQUESTED** [review]
I have a few notes, with one actual requested change and one question. The rest you can take or leave.

**Review by bthirion — COMMENTED** [review]
LGTM otherwise. Thx.

File: nilearn/glm/first_level/experimental_paradigm.py (hunk 4 above)
Comment (rob-luke) [reply nilearn/glm/first_level/experimental_paradigm.py:98]: I believe so because this test still passes after my change, but maybe there are other edge cases not tested?

https://github.com/nilearn/nilearn/blob/4555aaaa78e9baff7a2ff727c7ea30a98b753f00/nilearn/glm/tests/test_paradigm.py#L97-L101

File: nilearn/glm/tests/test_dmtx.py (hunk 1 above)
Comment (rob-luke) [reply nilearn/glm/tests/test_dmtx.py]: Done, thanks

File: nilearn/glm/tests/test_first_level.py [hunk 5]
```diff
@@ -33,14 +32,15 @@
...
+    mask, fmri_data, design_matrices =\
+        generate_fake_fmri_data_and_design(shapes, rk)
 
     # Give an unfitted NiftiMasker as mask_img and check that we get an error
     masker = NiftiMasker(mask)
     with pytest.raises(ValueError,
                        match="It seems that NiftiMasker has not been fitted."):
-        single_session_model = FirstLevelModel(mask_img=masker).fit(
-                fmri_data[0], design_matrices=design_matrices[0])
+        _ = FirstLevelModel(mask_img=masker).fit(
```
[the remark this answers is not in the document]
Comment (rob-luke) [reply nilearn/glm/tests/test_first_level.py]: I found a few more of these and adjusted them to match your suggestion.

**Comment by rob-luke** [comment]
Thanks for the reviews @bthirion and @tsalo. It looks good to merge from my perspective now, but let me know if there are any further changes you wish me to make. I heavily utilise this great GLM package, so I am motivated to help you maintain it.

**Review by tsalo — APPROVED** [review]
LGTM. Thanks!

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by bthirion** [comment]
Merging, then.
